app/controllers/home_controller.rb

```diff
@@ -2,4 +2,9 @@ class HomeController < ApplicationController
   def index; end
 
   def portfolio; end
+
+  def transaction
+    @transactions = current_buyer.transactions if current_buyer
+    @transactions = Transaction.where(broker_email: current_broker.email.to_s) if current_broker
```

Comment:
Why can't we use the `user_id` instead?

Reply:
broker email is used for the uniqueness. we also used devise's helpers to distinguish buyer and/or broker

Reply:
The transaction table already has user_id column used to store the buyer user id. That is why I chose to use the email address of the broker to distinguish the specific broker who brokered the deal.

Reply:
oww this is for the broker, then use `broker_id` instead.